jib-core/src/main/java/com/google/cloud/tools/jib/cache/LayerEntriesSelector.java

```diff
@@ -43,11 +43,13 @@
  *     "sourceFile": "source/file/for/layer/entry/1",
  *     "extractionPath": "/extraction/path/for/layer/entry/1"
  *     "lastModifiedTime": "2018-10-03T15:48:32.416152Z"
+ *     "permissions": 511
```

Comment:
What do you think about using the string here instead?

Reply:
I don't think this json is ever visible to anyone, so I didn't think it was worth it to add the extra step to make it more readable. But on the other hand, maybe we should add a method to `FilePermissions` to get the permission bits as a string anyway.

Reply:
I'd still say at this point it's fine as is, but I don't feel strongly about it.

Reply:
Strings might be better since 0511 would really be 329 here.

Reply:
Right, but it's 511 not 0511, so it would actually be equal to 0777. And again, since this json is never actually readable (I think it just get converted straight into a digest) I wasn't too worried about people mixing up decimal/octal. The only thing that really matters is whether or not the contents change, not what they specifically are.

Reply:
511 here, you probably meant the usual 777.

> it was worth it to add the extra step to make it more readable

It's very easy to write a string in octal with the static `Integer.toString(num, radix)` (the other way is easy too), so I vote for string.

Reply:
I didn't mean it's difficult to code, I meant the code is performing one less step this way, since all we're doing is writing out the internal representation of `FilePermissions` to json. It's an extra operation to convert it to a string, which I didn't think was useful since this json is never actually useful to a user.

>511 here, you probably meant the usual 777.

If `layerEntry.getPermissions().getPermissionBits()` is 0777, then the json will be 511, won't it?

Reply:
It seems strange to me to perform the extra step to make json that's immediately turned into a digest more readable, but I guess we can go with that if people feel strongly about it.

Reply:
> If layerEntry.getPermissions().getPermissionBits() is 0777, then the json will be 511, won't it?

Oh, sorry. I didn't really mean something is wrong. Was saying to @briandealwis just to say 511 is from 777.

> writing out the internal representation of `FilePermissions` to json

Ah, I didn't go over the code yet and didn't notice this is the result of a direct serialization.

Reply:
Looking at the `LayerEntrySelector.permissions` field, so the question is about whether to have the field of type `String` instead of `int`? I think `String` will work fine. Any issue?